include/dlaf/permutations/general/impl.h

```diff
@@ -231,7 +232,7 @@ void all2allData(common::Pipeline<comm::Communicator>& sub_task_chain, int nrank
         std::transform(recv_counts.cbegin(), recv_counts.cend(), recv_counts.begin(), mul_const);
 
         // Note: that was guaranteed to be contiguous on allocation
-        T* send_ptr = send_tiles[0].ptr();
+        const T* send_ptr = send_tiles_fut[0].get().ptr();
```

Comment:
@msimberg @rasolca just highlighting this, where I get a raw pointer from a tile in a shared_future.

AFAICT this should not be a problem since the shared_future is kept alive by `transfroMPI`.

Do you see any issue?

Reply:
This looks ok to me.